src/os/bluestore/BlueStore.cc

```diff
@@ -3904,6 +3886,26 @@ int BlueStore::_set_cache_sizes()
 	 << dendl;
     return -EINVAL;
   }
+
+  double cache_kv_min = cct->_conf->bluestore_cache_kv_min;
+  double cache_kv_min_ratio = 0;
+
+  // if cache_kv_min is negative, disable it
+  if (cache_size > 0 && cache_kv_min >= 0) {
+    cache_kv_min_ratio = std::max((double)cache_kv_min / (double)cache_size,
+				  (double)1.0);
+    if (cache_kv_min_ratio > cache_kv_ratio) {
+      dout(1) << __func__ << " kv_min_ratio " << cache_kv_min_ratio
+            << " > kv_ratio " << cache_kv_ratio
+            << dendl;
+      cache_meta_ratio = cache_meta_ratio + cache_kv_ratio - cache_kv_min_ratio;
```

Comment:
IMO that's not 100% valid formula.  E.g. for cache_meta_ratio = 0.3  and cache_kv_ratio = 0.2 and cache_kv_min_ratio = 1.0 it results in cache_meta_ratio = -0.5 and subsequent cache_data_ratio = 0.5

Reply:
BTW this formula also implies that additional space given to kv cache is taken from meta one. Preserving data cache ratio as is (to some degree), e.g. for the case cache_meta_ratio = 0.5 and cache_kv_ratio = 0.3 and cache_kv_min_ratio = 0.5 we'll get cache_meta_ratio decrease while cache_data_ratio is preserved at 0.2 .
Not sure that's the best policy.
Maybe something like 
cache_meta_ratio = min(cache_meta_ratio, 1.0 - cache_kv_min_ratio)
?